force-app/main/default/classes/ExchangeRateScheduler.cls

```
@@ -0,0 +1,243 @@
+            
+        } catch (Exception e) {
+            System.debug(LoggingLevel.ERROR, 'Failed to schedule exchange rate update: ' + e.getMessage());
+            throw e;
+        }
+    }
+    
+    /**
+     * @description Schedules the exchange rate update for weekdays only
+     * @return String Job ID
+     */
+    public static String scheduleWeekdayUpdate() {
+        try {
+            // Remove any existing scheduled jobs with the same name
+            removeExistingScheduledJobs();
+            
+            // Schedule new job for weekdays only
+            ExchangeRateScheduler scheduler = new ExchangeRateScheduler();
+            String jobId = System.schedule(JOB_NAME_PREFIX + ' (Weekdays) - ' + DateTime.now().format('yyyy-MM-dd HH:mm:ss'),
+            CRON_EXPRESSION_WEEKDAYS,
+            scheduler);
+            
+            System.debug('Weekday exchange rate update job scheduled with ID: ' + jobId);
+            return jobId;
+            
+        } catch (Exception e) {
+            System.debug(LoggingLevel.ERROR, 'Failed to schedule weekday exchange rate update: ' + e.getMessage());
+            throw e;
+        }
+    }
+    
+    /**
+     * @description Removes existing scheduled jobs with the same name pattern
+     */
+    private static void removeExistingScheduledJobs() {
+        try {
+            List<CronTrigger> existingJobs = [
+                SELECT Id, CronJobDetail.Name, State
+                FROM CronTrigger
+                WHERE CronJobDetail.Name LIKE :JOB_NAME_PREFIX + '%'
+                AND State IN ('WAITING', 'ACQUIRED', 'EXECUTING')
+            ];
+            
+            for (CronTrigger job : existingJobs) {
+                System.abortJob(job.Id);
+                System.debug('Aborted existing job: ' + job.CronJobDetail.Name);
+            }
+            
+        } catch (Exception e) {
+            System.debug(LoggingLevel.WARN, 'Error removing existing scheduled jobs: ' + e.getMessage());
```

🛑 **[Critical Robustness] Prevent Silent Failures in Job Management**

This `catch` block in `removeExistingScheduledJobs` is a ticking time bomb. If a user without permissions tries to schedule a job from the UI, this method will fail, but the exception gets swallowed. The code will then proceed to schedule a *new* job, leaving the old one still active.

This can easily lead to multiple duplicate jobs running in production, and no one will know until things start breaking. We need to re-throw the exception from this `catch` block to ensure the entire operation fails loudly and the error is surfaced to the user.